### ISSUE TITLE
Canvas view scaling error

### ISSUE THREAD

**Issue by juliomateoslangerak** [MicronOxford/cockpit#532]
I get this error with a black Image:
```
C:\Users\julio\PycharmProjects\cockpit\cockpit\gui\imageViewer\viewCanvas.py:188: RuntimeWarning: invalid value encountered in double_scalars
  return - (self.vmin - self.dmin) / (self.dptp * self.scale)
C:\Users\julio\PycharmProjects\cockpit\cockpit\gui\imageViewer\viewCanvas.py:344: RuntimeWarning: invalid value encountered in double_scalars
  return -1 + 2 * (val - self.lbound) / (self.ubound - self.lbound)
```

**Comment by mickp** [comment]
This is probably a divide-by-zero error which can happen when the first image is full of a single value.

Options:
 a) limit the the determinant;
 b) set it to 1 if it is exactly equal to zero.

(a) works well for ints (just limit it to 2^-(bit depth), but maybe not for floats.
(b) is equivalent to (a) if there's one bit of noise about a single value; for one or more bits of noise, it will scale the noise to fill the dynamic range. If the image is truly single-valued, we don't scale it, and it just appears as a single grey level (which it is).

I think I prefer (b)?

**Comment by mickp** [comment]
Fixed in 000b95a56de86291afbb587231b4389d05692768.
